src/post/content.ts

```diff
@@ -125,6 +141,18 @@ export class ContentPreparer {
         return content + PAID_CONTENT_PREVIEW_HTML(url);
     }
 
+    private addMentions(content: string, mentions: Mention[]) {
+        let preparedContent = content;
+        for (const mention of mentions) {
+            const mentionRegex = new RegExp(mention.name, 'g');
```

Comment:
Think we need to escape `mention.name` here, otherwise it will match more than we want.

Example: `mention.name` is `@bob@my.site.com`

`new RegExp('@bob@my.site.com', 'g');` will also incorrectly match `@bob@myXsiteYcom` (as `.` matches any characters)

Reply:
Great point. I also added a test for this and it failed 😅 Nice catch. Fixed here https://github.com/TryGhost/ActivityPub/pull/666/commits/ac30087a15cead75ae3a68a74bc6ae8eb5857d03